drivers/adc/adc_stm32.c

```diff
@@ -921,6 +957,9 @@ static int adc_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+	__ASSERT_NO_MSG(config->clock_prescaler < ARRAY_SIZE(adc_clock_prescaler));
```

Comment:
This is still a runtime assert. This could be found at compile time.

Reply:
changing to BUILD_ASSERT